src/fu-engine.c

```diff
@@ -3534,6 +3534,27 @@ fu_engine_get_device (FuEngine *self, const gchar *device_id, GError **error)
 	return device;
 }
 
+/**
+ * fu_engine_get_device_by_guid:
+ * @self: A #FuEngine
+ * @guid: A GUID
+ * @error: A #GError, or %NULL
+ *
+ * Gets a specific device.
+ *
+ * Returns: (transfer full): a device, or %NULL if not found
+ **/
+FuDevice *
+fu_engine_get_device_by_guid (FuEngine *self, const gchar *guid, GError **error)
+{
+	FuDevice *device;
+
+	device = fu_device_list_get_by_guid (self->device_list, guid, error);
```

Comment:
This makes an assumption there is only one device (it will potentially cause the wrong behavior if multiple are plugged in)

Reply:
Fixed I think.